### DISCUSSION TITLE
Unneeded lookups in DespawnRecursive

### DISCUSSION THREAD

**Discussion by xaoctech** [bevyengine/bevy#2274] [General]
Why `despawn_with_children_recursive_inner` internally uses `despawn_with_children_recursive` to despawn its children (instead of `despawn_with_children_recursive_inner`? It seems there is no need to remove it from the parent (parent children are already taken). And a situation when Parent and Children components are out of sync should trigger an assertion IMO.

**Comment by mockersf** [comment]
I agree with you so I opened #2278 👍 

Thanks for bringing that up!
